adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -18,12 +18,25 @@ private void generateNotices(MockValidationResultRepo mockValidationResultRepo,
     }
 
     @Test
-    public void creationOf100NoticeShouldNotRaiseException() {
+    public void creationOf100NoticeShouldNotExceedMemoryLimit() {
+
+        long totalMemory = Runtime.getRuntime().totalMemory(); // bytes
+
+        InMemoryValidationResultRepository resultRepository = new InMemoryValidationResultRepository();
+
+        generateNotices(resultRepository, 1);
+        resultRepository.getAll();
 
         long freeMemory = Runtime.getRuntime().freeMemory();  // bytes
 
-        generateNotices(mockValidationResultRepo, 100);
-        mockValidationResultRepo.getAll();
+        System.out.println(
+                String.format("Generating 100 notices: Total memory: %s megabytes, Free memory: %s megabytes, Used memory: %s megabytes",
+                        totalMemory / 1_000_000,
+                        freeMemory / 1_000_000,
+                        (totalMemory - freeMemory) / 1_000_000)
+        );
+
+        assert (totalMemory - freeMemory < 455_000_000);
```

Comment:
Regarding this 455_000_000  value, I am not sure how the memory process works. Logically generating 100 notices should consume less memory than generating 1000. 
But the order in which the tests are executed makes me wonder how the JVM memory works, does it extend at some point?  On the screenshot, you can see that the test are not executed in the same order they are written. 

After running the test for 2 000 000 notices, it looks like we are using "another memory" (total value increases). I might be wrong thinking that total memory is the total memory allowed by the JVM. The documentation for Runtime.totalMemory() indicates:

>      * Returns the total amount of memory in the Java virtual machine.


![Capture d’écran, le 2020-03-19 à 12 22 58](https://user-images.githubusercontent.com/35747326/77091791-62892500-69df-11ea-908a-58d703ac39a0.png)

Reply:
Java does dynamically resize memory as it executes (up to a limit imposed by the OS), as well as reclaims member and discards unused objects (garbage collection). This seems to be a good short explanation - https://stackify.com/java-heap-vs-stack/. Note that there is a `maxMemory()` method too which should give you the limit.

For our purposes in benchmarking here, we really care the most about heap memory, which is where objects are stored - we want to measure how much space entities are taking up.  From a quick look at the Java docs, it seems like the methods you're using are for the heap.

Before and after each test you should call `Runtime.gc()` to ask the garbage collector to run and get rid of any unused objects to avoid tests having an impact on each other. You should be able to do this via `@Before` and `@After` annotated methods. Try that and see what the numbers show

Reply:
Very interesting explanation, i'll adapt and see what happens. Thanks

Reply:
After checking the resource, since we use JUnit5, @Before and @After become @BeforeEach and @AfterEach